examples/gcoap/Makefile

```diff
@@ -0,0 +1,63 @@
+# Default Makefile, for host native networking
+
+# name of your application
+APPLICATION = gcoap
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+BOARD_INSUFFICIENT_MEMORY := chronos msb-430 msb-430h nucleo-f030 nucleo-f334 \
+                             stm32f0discovery telosb weio wsn430-v1_3b wsn430-v1_4 \
+                             z1
+
+# Must read nordic_softdevice_ble package before nanocoap package. However,
+# can't read it explicitly here because it is read later, as a dependency for
+# the nrf52dk board.
+BOARD_BLACKLIST := nrf52dk
+
+# Redefine port, for example use 61616 for RFC 6282 UDP compression.
+#GCOAP_PORT = 5683
+#CFLAGS += -DGCOAP_PORT=$(GCOAP_PORT)
+
+# Redefine request token length, max 8.
+#GCOAP_TOKENLEN = 2
+#CFLAGS += -DGCOAP_TOKENLEN=$(GCOAP_TOKENLEN)
+
+USEPKG += nanocoap
+# Required by nanocoap, but only due to issue #5959.
+USEMODULE += posix
+# Required by nanocoap to compile nanocoap_sock.
+USEMODULE += gnrc_sock_udp
+
+# Include packages that pull up and auto-init the link layer.
+# NOTE: 6LoWPAN will be included if IEEE802.15.4 devices are present
+USEMODULE += gnrc_netdev_default
+USEMODULE += auto_init_gnrc_netif
+# Specify the mandatory networking modules
+USEMODULE += gnrc_ipv6_router_default
```

Comment:
Maybe you can remove the router requirements on this, too? RPL you already commented out below anyways ;-) (see #6007).